daemon/src/fan_out.rs

```diff
@@ -1,32 +1,92 @@
+use anyhow::Result;
 use xtra::prelude::MessageChannel;
 
 /// A fan-out actor takes every incoming message and forwards it to a set of other actors.
-pub struct Actor<M: xtra::Message<Result = ()>> {
+pub struct Actor<M: xtra::Message<Result = Result<()>>> {
     receivers: Vec<Box<dyn MessageChannel<M>>>,
 }
 
-impl<M: xtra::Message<Result = ()>> Actor<M> {
+impl<M: xtra::Message<Result = Result<()>>> Actor<M> {
     pub fn new(receivers: &[&dyn MessageChannel<M>]) -> Self {
         Self {
             receivers: receivers.iter().map(|c| c.clone_channel()).collect(),
         }
     }
 }
 
-impl<M> xtra::Actor for Actor<M> where M: xtra::Message<Result = ()> {}
+impl<M> xtra::Actor for Actor<M> where M: xtra::Message<Result = Result<()>> {}
 
 #[async_trait::async_trait]
 impl<M> xtra::Handler<M> for Actor<M>
 where
-    M: xtra::Message<Result = ()> + Clone + Sync + 'static,
+    M: xtra::Message<Result = Result<()>> + Clone + Sync + 'static,
```

Comment:
Is this change necessary?

Reply:
I don't think it's strictly necessary, but I think it goes with the theme of phasing out `log_error`:

1. We want to, in general, shift from logging at the handler to returning errors to the sender.
2. The `fan_out::Actor` was preventing us from passing errors through back to the sender.

I think it would be better if we could fan out messages of both types here i.e. `()` and `Result<()>`. Is there any way we could do that?

Reply:
> We want to, in general, shift from logging at the handler to returning errors to the sender.

Yes but in this case, `fan_out` was explicitly designed to only work with messages that never fail to avoid exactly the decision making of "what do we do if one of the messages fails?"

So unless this is a change motivated by another change that is more related to the `log_error` macro, I'd leave that in place for now.